packages/runtime/src/execute/expression.ts

```diff
@@ -125,11 +130,15 @@ export const wrapOperation = (
 const prepareJob = async (
   expression: string | Operation[],
   context: Context,
-  opts: Options = {}
+  opts: Options = {},
+  moduleOverrides: ModuleInfoMap = {}
 ): Promise<JobModule> => {
   if (typeof expression === 'string') {
     const exports = await loadModule(expression, {
       ...opts.linker,
+      // allow module paths and versions to be overriden from the defaults
+      // TODO I think this is too harsh and path information will be lost
+      modules: Object.assign({}, opts.linker?.modules, moduleOverrides),
```

Comment:
This isn't a good solution because it totally blats the base modules object.

Ok, maybe it's academic because of how the CLI and Engine work. But really this should be a merge rather than a blat.

The problem is that if the linker options includes a path for a module, the path is lost if the override contains a version. It's silly really.

just needs a better merge algorithm. The hardest bit about that is that this is hard code to test right now...